noxfile.py

```diff
@@ -1,9 +1,50 @@
+from pathlib import Path
+
 import nox
 
-@nox.session(python=["3.11"]) # The python version should match the readthedocs configuration.
+requirements_directory = Path("requirements").resolve()
+
+requirements_files = [
+    requirements_input_file_path.stem
+    for requirements_input_file_path in requirements_directory.glob("*.in")
+]
+
+
+@nox.session(name="pip-compile", python=["3.11"])
+@nox.parametrize(["req"], arg_values_list=requirements_files, ids=requirements_files)
+def pip_compile(session: nox.Session, req: str):
+    """Generate lock files from input files or upgrade packages in lock files."""
+    # fmt: off
+    session.install(
+      "-r", f"{requirements_directory}/pip-tools.in",
+      "-c", f"{requirements_directory}/pip-tools.txt",
```

Comment:
I wonder if nox would stomach native pathlib objects as CLI arg segments...
```suggestion
      "-r", requirements_directory / "pip-tools.in",
      "-c", requirements_directory / "pip-tools.txt",
```

Reply:
Only rather new nox versions support this. It'd be better to coerce it to a string first.

Reply:
@gotmax23 I've had a go at converting the requirements_directory var to a string. It took me a minute because that lead to type error with the `/` operator. I found the `join()` function was necessary to combine the directory and file name. It seems a bit more complicated now though. Please take a look. Thanks.

Reply:
Cheers @gotmax23 I don't know why I did that in such a crappy way. Thank you for the suggestions.

Reply:
@oraNod the idea is that you use `pathlib` objects internally everywhere. And then, when you need to communicate path to programs external to yours, you only convert that to a string at a boundary between programs. And calling sub-processes is one of such placess. So instead of concatenating two strings with a path separator that is specific to some operating systems, you could work with the native pathlib objects, produce subfolders as pathlib objects (through `/`) and only convert the result when (and if!) needed, by doing `str(path_thingy)`.

Reply:
*(oh, I see you did that already — well done!)*

Reply:
Thanks for the explanation @webknjaz - credit for the _well done_ should likely go to @gotmax23 - I at least made an attempt and was in the right ballpark so I guess there's that.